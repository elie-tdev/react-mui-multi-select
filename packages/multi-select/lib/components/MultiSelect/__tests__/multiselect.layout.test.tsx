import React from 'react'
import { render } from '@testing-library/react'
import { MultiSelect } from '..'
import { top100Films, Film } from '../mockdata'

describe('MultiSelect', () => {
  it('should render successfully: ', () => {
    const getOptionLabel = (u: Film) => u.title
    const { container } = render(
      <MultiSelect
        options={top100Films}
        getOptionLabel={getOptionLabel}
        label="Films"
        placeholder="Braveheart"
      />,
    )
    expect(container).toMatchSnapshot()
    expect(container).toBeTruthy()
  })
})
