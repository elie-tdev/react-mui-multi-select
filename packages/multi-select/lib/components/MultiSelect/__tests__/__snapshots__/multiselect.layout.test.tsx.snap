// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect should render successfully:  1`] = `
<div>
  <div
    class="MuiBox-root css-1vkq819"
  >
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
      style="width: 100%;"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Films
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1a1fmpi-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id=":r0:"
            placeholder="Braveheart"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
